### PR TITLE
Use flat-token

### PR DESCRIPTION
**Pull request by pczarn**

### CODE REVIEW

File: src/proc_macro.rs [hunk 1]
```diff
@@ -168,81 +162,38 @@ impl<S: AsRef<str>, Pats: Deref<Target = [FlatTokenPat<S>]>> fmt::Debug for Pat<
...
-                out.push(FlatToken::Delim(open, span));
-                flatten(stream, out);
-                FlatToken::Delim(close, span)
-            }
-            TokenTree::Ident(tt) => FlatToken::Ident(tt),
-            TokenTree::Punct(tt) => FlatToken::Punct(tt),
-            TokenTree::Literal(tt) => FlatToken::Literal(tt),
-        };
-        out.push(flat);
+pub fn matches_pat(this: &FlatToken, pat: &FlatTokenPat<impl AsRef<str>>) -> bool {
```
Comment (eddyb) [inline src/proc_macro.rs]: You can make this and extract methods on `FlatTokenPat`.

Comment (pczarn) [reply src/proc_macro.rs]: Done.

File: Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "grammer"
-version = "0.0.1"
+version = "0.0.2"
```
Comment (eddyb) [inline Cargo.toml]: Oh I forgot to mention, please don't change the version number, `grammer` is still unpublished.

Comment (pczarn) [reply Cargo.toml]: Done.

File: src/proc_macro.rs [hunk 3]
```diff
@@ -200,8 +189,8 @@ impl FlatToken {
         }
     }
 
-    pub fn matches_pat(&self, pat: &FlatTokenPat<impl AsRef<str>>) -> bool {
-        match (self, pat) {
+    pub fn matches_pat(&self, ft: &FlatToken) -> bool {
```
Comment (eddyb) [inline src/proc_macro.rs]: This can be just `matches`.

File: src/proc_macro.rs [hunk 4]
```diff
@@ -168,20 +163,14 @@ impl<S: AsRef<str>, Pats: Deref<Target = [FlatTokenPat<S>]>> fmt::Debug for Pat<
...
-            FlatToken::Ident(tt) => tt.span(),
-            FlatToken::Punct(tt) => tt.span(),
-            FlatToken::Literal(tt) => tt.span(),
-        }
-    }
-
-    pub fn extract_pat(&self) -> FlatTokenPat<String> {
-        match self {
+impl<S: AsRef<str>> FlatTokenPat<S> {
+    pub fn extract_pat(ft: &FlatToken) -> FlatTokenPat<S>
```
Comment (eddyb) [inline src/proc_macro.rs]: This can be just `extract`.

**Review by eddyb — APPROVED** [review]

**Pull request merged**